src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1460,14 +1469,30 @@ def aks_create(cmd,     # pylint: disable=too-many-locals,too-many-statements,to
                 attach_acr,
                 headers,
                 no_wait)
-            return created_cluster
+            break
         except CloudError as ex:
             retry_exception = ex
             if 'not found in Active Directory tenant' in ex.message:
                 time.sleep(3)
             else:
                 raise ex
-    raise retry_exception
+    else:
+        raise retry_exception
+
+    # Create the DCR Association here, it must be done after the cluster is created.
+    if monitoring and enable_msi_auth_for_monitoring:
+        try:
```

Comment:
Not sure if there's a more concise way to wait for cluster creation to finish (since the object created_cluster can be a waitable type or not depending on if --no-wait is passed). Open to suggestions here.

Reply:
When the condition `monitoring and enable_msi_auth_for_monitoring` is met and then call method `_put_managed_cluster_ensuring_permission()`, how about fixing parameter `no_wait` to False in this situation?
Because in this case,  `no_wait=True` is meaningless and ineffective

Reply:
Makes sense. An error is thrown if --attach-acr, --enable-managed-identity, and --no-wait are all three specified, should I follow the same pattern or silently disable --no-wait?